backend/copr_backend/sign.py

```diff
@@ -167,7 +167,11 @@ def sign_rpms_in_dir(username, projectname, path, chroot, opts, log):
     try:
         get_pubkey(username, projectname, log, opts.sign_domain)
     except CoprSignNoKeyError:
-        create_user_keys(username, projectname, opts)
+        # We don't want to try in indefinitely because that could lead to
+        # forver stuck builds but we can afford trying to create the key for
+        # several hours
+        timeout = 60 * 60 * 10
+        create_user_keys(username, projectname, opts, timeout=timeout)
```

Comment:
This is weird.  What is the default behavior without the timeout, would that wait indefinitely?  Why not wait indefinitely till the keygen is fixed (it needs to be fixed sooner or later)?  Perhaps I just need a cup of coffee :-)

Reply:
Btw., since we create the key -> I'd expect that we should re-try doing the `get_pubkey` right after that.  Doesn't this lead to the situation without `pub` key in resultdir?  If not, can you in-line comment why? :)

Reply:
So, the default behavior without specifying timeout is this:

```python
class SafeRequest:
    def __init__(..., try_indefinitely=False, timeout=2 * 60):
```

>  I'd expect that we should re-try doing the get_pubkey right after that.

IMHO the `get_pubkey` is only a check and we don't use its value. It is just important, that the key exists when `_sign_one` is called.

Now ... if we do a several hours timeout as I am currently proposing, it could still fail that the key is missing, as it fails now. But after several hours, what is the chance? If we would do `try_indefinitely=True` instead of a timeout, we can be sure that the key will exist.

If you are not worried that `try_indefinitely=True` might cause forever running builds for some obscure reason, we can do that instead :-)

Reply:
Ah, there's the default 120s limit, I missed that.  Yes, then what you propose is better definitely, and I am also for trying `try_indefinitely` :-) indefinitely running job would only force us to observe and fix other bugs.